mssqlcli/jsonrpc/tests/test_jsonrpcclient.py

```diff
@@ -188,11 +189,27 @@ def test_receive_invalid_response_exception(self):
             self.assertEqual(
                 str(exception),
                 u'Content-Length was not found in headers received.')
+
+            # flaky test fix: logic that waits until threads shut down
+            count = 0
+            while count < 5:
+                active_threads = threading.enumerate()
+                if test_client.request_thread not in active_threads or \
+                    test_client.response_thread in active_threads:
+                    time.sleep(5)
```

Comment:
This is 5 seconds, right? Are there any timeouts on the tests themselves?

Reply:
Yes, 5 seconds. No timeouts from what I can tell.